keyboards/z12/keymaps/zigotica/keymap.c

```diff
@@ -16,25 +16,75 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include "zigotica.h"
+#include "raw_hid.h"
+
+#ifdef RAW_ENABLE
+void raw_hid_receive(uint8_t* data, uint8_t length) {
+    layer_clear();
+    if (data[0] == 99) {
+        layer_on(_BASE);
+    }
+    else {
+        layer_on(data[0]);
+    }
+}
+#endif
 
 // Custom Keycodes
-#define MODE_1 TO(_TERMINAL)
+#define MODE_1 TO(_BASE)
 #define MODE_2 TO(_FIGMA)
 #define MODE_3 TO(_BROWSER)
 #define MODE_4 TO(_VIM)
 
 enum custom_keycodes {
-    VIM_SIP = SAFE_RANGE
+    VIM_SIF = SAFE_RANGE,
+    VIM_SIP,
+    VIM_RIF,
+    VIM_RIP,
+    VIM_NEW
 };
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
     switch (keycode) {
-        case VIM_SIP:
+        case VIM_SIF:// Search in File
+            if (record->event.pressed) {
+                register_code(KC_ESC);
+                tap_code(KC_SLASH);
+            } else { // released
+                unregister_code(KC_ESC);
+            }
+        break;
+        case VIM_SIP:// Search in Project
             if (record->event.pressed) {
                 register_code(KC_ESC);
                 SEND_STRING(":Ag ");
-            } else {
-                // released
+            } else { // released
+                unregister_code(KC_ESC);
+            }
+        break;
+        case VIM_RIF:// Replace in File
+            if (record->event.pressed) {
+                register_code(KC_ESC);
+                SEND_STRING(":%s/a/b/g");
+            } else { // released
+                unregister_code(KC_ESC);
+            }
+        break;
+        case VIM_RIP:// Replace in Project
+            if (record->event.pressed) {
+                register_code(KC_ESC);
+                SEND_STRING(":cdo %s/a/b/g");
+            } else { // released
+                unregister_code(KC_ESC);
+            }
+        break;
+        case VIM_NEW:// New buffer
+            if (record->event.pressed) {
+                register_code(KC_ESC);
+                SEND_STRING(":vnew");
+                register_code(KC_ENT);
```

Comment:
Optionally can be replaced with:
```suggestion
                SEND_STRING("\e:vnew\n");
```
etc.

Reply:
I like that, thank you!